.circleci/bin/deploy-ci.py

```diff
@@ -489,7 +491,33 @@ def cli(
                 dry_run=dry_run,
             )
 
+            if environment == "prod":
+                # check healthcheck post-deploy
+                wait_time = time.time() + (5 * 60)
+                while time.time() < wait_time:
+                    # throttle the amount of logs and request load during startup
+                    time.sleep(30)
+
+                    # this resets on each loop since we only care about the last run
+                    failed_post_check = False
+
+                    health_check = ssh(
```

Comment:
We could also just issue a request from circleci to the node directly instead of ssh-ing in. I think both are valid options, just wanted to point out.

Reply:
Sorry, I don't follow.

Do you mean a `curl` on `/health_check` from CircleCI?

Reply:
yea through the DNS. it would be further from the code but also diagnose if we screw up some port map or something in a release

Reply:
Ah, okay! Thanks for the explanation!

That would involve a tighter coupling between deploy-ci.py and CircleCI that would make orchestration a bit harder to follow and debug.

Going that route would also make Sid's request for a [Break-Glass flow](https://www.notion.so/audiusproject/Automated-Production-Deployments-67f4fb5fb7574320a45dd63082ec4adf#ffecaa2f703a4a86a3b64e49b0f17ac6) potentially more complex. Given last night's CircleCI outage between our deployments, I'd strongly vote to not having our deployments tightly coupled with CircleCI.

Plus, ideally [cluster health would be decoupled from the deployment](https://www.notion.so/audiusproject/Automated-Production-Deployments-67f4fb5fb7574320a45dd63082ec4adf#08062633007946baabfbc62441f3ae4c) (as much as possible) by way of Grafana Alerts.